### PR TITLE
feat: wire buy action home

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1152

- Wire buy sell order action on home page
- Refactor featured projects to its own component
- Refactor buy sell order action on projects page into `BuySellOrderFlow`to make it reusable
- Revert sort on home page to keep projects on the featured section with sell orders first

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [x] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1. Go to https://deploy-preview-1175--regen-registry.netlify.app/

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-registry/src/pages/Projects/hooks/useProjectsSellOrders.tsx [hunk 1]
```diff
@@ -52,42 +55,46 @@ const getProjectDisplayData = async (
...
-      const purchaseInfo = getPurchaseInfo(project.id, sellOrders);
-      let metadata;
-      if (project.metadata.length) {
-        try {
-          metadata = await getMetadata(project.metadata);
-        } catch (error) {
-          // eslint-disable-next-line
-          console.error(error);
+    projects
+      .sort(sortProjectsBySellOrdersAvailability(sellOrders))
```
Comment (flagrede) [inline web-registry/src/pages/Projects/hooks/useProjectsSellOrders.tsx:59]: @mhagel I double check this and it is still needed. If you want to check you can try giving a limit lower to the number of projects with sell orders + metadata.

File: web-registry/src/features/marketplace/BuySellOrderFlow.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import { ProjectWithOrderData } from 'pages/Projects/Projects.types';
+import { Link } from 'components/atoms';
+import { BuyCreditsModal } from 'components/organisms';
+import { useMsgClient } from 'hooks';
+
+type Props = {
+  selectedProject: ProjectWithOrderData | null;
+};
+
+export const BuySellOrderFlow = ({ selectedProject }: Props): JSX.Element => {
```
Comment (flagrede) [inline web-registry/src/features/marketplace/BuySellOrderFlow.tsx:24]: This is existing code moved into a reusable component.

File: web-registry/src/pages/Home/Home.FeaturedProjects.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+export const FeaturedProjects = ({
+  projects,
+  isLoading,
+}: Props): JSX.Element => {
+  const styles = useHomeStyles();
+  const navigate = useNavigate();
+  const [selectedProject, setSelectedProject] =
+    useState<ProjectWithOrderData | null>(null);
+
+  return (
```
Comment (flagrede) [inline web-registry/src/pages/Home/Home.FeaturedProjects.tsx:33]: This is existing code moved into a separate component for readability.

File: web-registry/src/pages/Projects/hooks/useProjectsSellOrders.utils.tsx [hunk 4]
```diff
@@ -38,3 +39,18 @@ export const getPurchaseInfo = (
     },
   };
 };
+
+export const sortProjectsBySellOrdersAvailability =
```
Comment (flagrede) [inline web-registry/src/pages/Projects/hooks/useProjectsSellOrders.utils.tsx:43]: Reverted code.

**Comment by flagrede** [comment]
@erika testing link for buy sell order on home page: https://deploy-preview-1175--regen-registry.netlify.app/

**Review by haveanicedavid — APPROVED** [review]
lgtm

File: web-registry/src/features/marketplace/BuySellOrderFlow.tsx (hunk 2 above)
Comment (mhagel) [reply web-registry/src/features/marketplace/BuySellOrderFlow.tsx:24]: cool

File: web-registry/src/pages/Projects/hooks/useProjectsSellOrders.utils.tsx [hunk 5]
```diff
@@ -38,3 +39,18 @@ export const getPurchaseInfo = (
...
+  (sellOrders: SellOrderInfo[]) =>
+  (projectA: ProjectInfo, projectB: ProjectInfo) => {
+    const ordersForProjectA = sellOrders.filter(order =>
+      order.batchDenom.startsWith(projectA?.id),
```
Comment (mhagel) [inline web-registry/src/pages/Projects/hooks/useProjectsSellOrders.utils.tsx:47]:
```suggestion
      order.batchDenom.startsWith(`${projectA?.id}-`),
```
An edge case, but I think we need to do this so it doesn't pick up other project IDs, for example C01-111 vs C01-1111.

File: web-registry/src/pages/Projects/hooks/useProjectsSellOrders.utils.tsx [hunk 6]
```diff
@@ -38,3 +39,18 @@ export const getPurchaseInfo = (
...
+    );
+    const ordersForProjectB = sellOrders.filter(order =>
+      order.batchDenom.startsWith(projectB?.id),
```
Comment (mhagel) [inline web-registry/src/pages/Projects/hooks/useProjectsSellOrders.utils.tsx:50]:
```suggestion
      order.batchDenom.startsWith(`${projectB?.id}-`),
```

File: web-registry/src/pages/Projects/hooks/useProjectsSellOrders.tsx (hunk 1 above)
Comment (mhagel) [reply web-registry/src/pages/Projects/hooks/useProjectsSellOrders.tsx:59]: sounds good

**Review by mhagel — APPROVED** [review]
One edge case to maybe consider. Pre-approving

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**